cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/distributed/SparkExecutionService.java

```diff
@@ -69,7 +69,7 @@
   private static final Logger LOG = LoggerFactory.getLogger(SparkExecutionService.class);
   private static final Gson GSON = new Gson();
   private static final Type TOKEN_TYPE = new TypeToken<Map<String, String>>() { }.getType();
-  private static final long SHUTDOWN_WAIT_SECONDS = 30L;
+  private static final long SHUTDOWN_WAIT_SECONDS = 20L;
```

Comment:
this happens to be the same wait as twill's shutdown wait, which causes really confusing behavior as this service will begin shutting down and then twill may or may not kill it partway through. So making this a little shorter to force shutdown a little earlier.